proptest/CHANGELOG.md

```diff
@@ -1,3 +1,20 @@
+## Unreleased
+
+### Breaking Changes
+
+- `prop_flat_map` can only be used with closures that produce a strategy that
+  implements `Copy`.
```

Comment:
I'm concerned about this, I often use Strategies over String, sometimes even `Just(String)`, which is `Clone` but not `Copy`.

Reply:
Is that perhaps a typo? The initial issue mentions `Clone`.

Regardless, it seems a little risky to require `Clone` as it *might* permanently break lots of proptests out there. Would it make sense to instead provide it under a new name, preserving the current `prop_flat_map`? Not ideal either, I know...

I otherwise am happy to see some effort in trying to do something about the shrinking behavior for `prop_flat_map`. Like @nzeh I've run into the issue of poor shrinking many times.

Reply:
Ugh.  Yes, that's a typo.  The updated implementation in the pull request requires `Clone`able strategies, not ones that implement `Copy`.  I aggree with @Andlon that adding even the requirement that the strategy implement `Clone` is suboptimal.  After spending quite some time thinking about how to avoid it at the time, I concluded that doing what my shrink implementation does (shrink the outer strategy first and then the inner one) is not possible with the current architecture of the `ValueTree` trait.  I *think* it is possible if we completely overhaul the architecture of at least parts of proptest, but that was not something I intended to undertake at the time.  Moreover, this overhaul in itself could lead to breakage of code that uses proptest because it would involve changing at least the `ValueTree` trait.